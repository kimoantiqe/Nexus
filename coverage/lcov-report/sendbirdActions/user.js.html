<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for sendbirdActions/user.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">sendbirdActions</a> user.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.22% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.22% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import SendBird from 'sendbird';
&nbsp;
const APP_ID = '5684560F-4764-4763-A8EB-3AD372D8B9EB';
&nbsp;
//Connect Function.
export const sbConnect = (userId, nickname) =&gt; {
    return new Promise((resolve, reject) =&gt; {
        //Create new connection.
        const sb = new SendBird({ 'appId': APP_ID });
        //Initiate connection to our sendbird backend
        sb.connect(userId, (user, error) =&gt; {
            //Handles the error
            <span class="missing-if-branch" title="else path not taken" >E</span>if (error) {
                reject('SendBird Login Failed.');
            } else {
                //Update the user profile to view his name, 
                //Second option is set to null until we implement 
                //profile pics.
<span class="cstat-no" title="statement not covered" >                sb.updateCurrentUserInfo(nickname, null, <span class="fstat-no" title="function not covered" >(u</span>ser, error) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >                    if (error) {</span>
<span class="cstat-no" title="statement not covered" >                        reject('Update User Failed.');</span>
                    } else {
<span class="cstat-no" title="statement not covered" >                        resolve(user);</span>
                    }
                })
            }
        })
    })
};
&nbsp;
//Disconnect Function.
export const sbDisconnect = () =&gt; {
    return new Promise((resolve, reject) =&gt; {
        const sb = SendBird.getInstance();
        <span class="missing-if-branch" title="else path not taken" >E</span>if (sb) {
            sb.disconnect(() =&gt; {
                resolve(null);
            });
        } else {
<span class="cstat-no" title="statement not covered" >            reject('Could not get instance');</span>
        }
    })
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 19 2018 21:32:30 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
