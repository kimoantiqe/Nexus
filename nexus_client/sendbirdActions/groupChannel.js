import SendBird from 'sendbird';

//Function to create 1-1 GroupChannel.
export const sbCreateChannel = (userIds, channelName) => {
    return new Promise((resolve, reject) => {
        const sb = SendBird.getInstance();
        if(sb){
            sb.GroupChannel.createChannelWithUserIds(userIds, true, channelName, null, null, null, (groupChannel, error) => {
                if (error) {
                    console.log('Create Channel failed');
                    console.log(error);
                    reject('SendBird Group Channel Creation failed');
                } else {
                    console.log('Create Channel worked');
                    resolve(groupChannel);
                }
            });
        } else {
            reject('There is no SendBird instance!');
        }
    });
};

export const sbCreateGroupChannelListQuery = () => {
    const sb = SendBird.getInstance();
    if(sb)
        return sb.GroupChannel.createMyGroupChannelListQuery();
    else    
        console.log("Group Channel Fail");
}

//Function to get a list of all the GroupChannels.
export const sbGetChannels = (channelListQuery) => {
    return new Promise((resolve, reject) => {
        channelListQuery.next(function(channelList, error){
            if (error) {
                reject("Could not retrieve the channel!");
            } else {
                resolve(channelList);
            }
        });        
    });
};

//Create a leave function so that when the user deletes the chat.
