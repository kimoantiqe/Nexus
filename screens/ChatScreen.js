import React, { Component } from 'react'
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  Button,
  TextInput,
  TouchableHighlight,
  FlatList,
  Dimensions,
  KeyboardAvoidingView
} from 'react-native'
import moment from 'moment'
import SendBird from 'sendbird'

import {
    Header,
    Left,
    Right,
    Body,
    Title
  } from "native-base";

import {Icon} from 'react-native-elements'
const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

console.disableYellowBox = true

var sb;

const themeColor = '#2f1959'
const themeDarkColor = '#287277'

var messageQuery;

export default class ChatScreen extends Component {
    
    static navigationOptions = {
        title: 'ChatScreen',
        header: null,
    };
    
    constructor(props) {
        super(props)
        this.getGroupChannel = this.getGroupChannel.bind(this)
        this.sendMessage = this.sendMessage.bind(this)
        this.state = { text: '', messages: [] , groupChannel: Object, friendName: '', userId: '', channelUrl: '', sb: Object }
        this.state.sb = SendBird.getInstance();
    }

    componentDidMount() {
        this.getGroupChannel();
        const ChannelHandler = new this.state.sb.ChannelHandler()
        ChannelHandler.onMessageReceived = (receivedChannel, message) => {
            if (receivedChannel.url === groupChannel.url) {
                const messages = []
                messages.push(message)
                const newMessages = messages.concat(this.state.messages)
                this.setState({ messages: newMessages })
                if (this.state.groupChannel.channelType == 'group') {
                    this.state.groupChannel.markAsRead()
                }
            }
        }
        this.state.sb.addChannelHandler('ChatScreen', ChannelHandler)

        const ConnectionHandler = new this.state.sb.ConnectionHandler()
        ConnectionHandler.onReconnectSucceeded = () => {
            this.getChannelMessage(true)
            channel.refresh()
        }
        this.state.sb.addConnectionHandler('ChatView', ConnectionHandler)
    }

    getGroupChannel() {
        this.state.channelUrl = this.props.navigation.getParam('channelUrl', null);
        this.state.userId = this.props.navigation.getParam('userID', null);
        //get groupchannel
        this.state.sb.GroupChannel.getChannel(this.state.channelUrl, (channel, error) => {
            //add alert and go back on press
            if(error){
                console.log('Could not get channel');
                console.log(error);
            }
            
            messageQuery = channel.createPreviousMessageListQuery();
            
            this.state.groupChannel = channel;
            
            if(this.state.userId === null)
                this.state.friendName = channel.name;            
            else 
                this.state.friendName = channel.members[0].userId === this.state.userId ? channel.members[1].nickname : channel.members[0].nickname;
               
            this.getChannelMessage(false);
        })
    }

    sendMessage() {
        if(this.state.text === "")
            return
        this.state.groupChannel.sendUserMessage(this.state.text, '', (message, error) => {
            if (error) {
                console.error(error)
                return
            }
            const messages = [].concat([message]).concat(this.state.messages)
            this.state.text = "";
            this.state.messages = messages;
        })
    }

    getChannelMessage(refresh) {
        if (refresh) {
            messageQuery = this.state.groupChannel.createPreviousMessageListQuery()
            this.state.messages = []
        }

        if (messageQuery) {
            if (!messageQuery.hasMore) {
                return
            }
            messageQuery.load(20, false, (response, error) => {
            if (error) {
                console.log('Get Message List Fail.', error)
                return
            }

            const messages = []
            for (let i = 0; i < response.length; i++) {
                messages.push(response[i])
            }

            const newMessageList = this.state.messages.concat(messages.reverse())
            this.setState({ messages: newMessageList })
            })
        }
    }

    renderMessage(msg) {
        const createdAt = moment(msg.createdAt).format('H:mm')
        if (msg._sender.userId === this.state.userId) {
            return (
                <View style={[styles.messageContainer, { paddingLeft: 50 }]}>
                <View style={{ justifyContent: 'flex-end' }}>
                <Text style={styles.createdAt}>{createdAt}</Text>
                </View>
                <View style={[styles.messageBody, { backgroundColor: themeColor }]}>
                <Text style={{ color: 'white' }}>{msg.message}</Text>
                </View>
                <Image source={{ uri: msg._sender.profileUrl }} style={[styles.profile, { marginLeft: 10 }]} />
                </View>
            )
        }
        return (
            <View style={[styles.messageContainer, { paddingRight: 50 }]}>
                <Image source={{ uri: msg._sender.profileUrl }} style={[styles.profile, { marginRight: 10 }]} />
                <View style={[styles.messageBody, { backgroundColor: '#efefef' }]}>
                    <Text style={{ color: 'black' }}>{msg.message}</Text>
                </View>
                <View style={{ justifyContent: 'flex-end' }}>
                    <Text style={styles.createdAt}>{createdAt}</Text>
                </View>
            </View>
        )
    }

    render() {
        return (
        <React.Fragment>
            <Header  iosBarStyle='light-content' androidStatusBarColor='#ffffff' style={styles.header}>
                    <Left/>
                        <Body>
                            <Title style={styles.headerTitle}>{this.state.friendName}</Title>
                        </Body>
                    <Right />
            </Header>
            <FlatList
                inverted
                onEndReached={() => this.getChannelMessage(false)}
                data={this.state.messages}
                renderItem={({ item }) => this.renderMessage(item)}
                style={{ flex: 1 }}
                />
            <KeyboardAvoidingView
                style={styles.container}
                behavior="padding"
            >
                <View style={{ flexDirection: 'row' }}>
                    <TextInput
                        style={styles.textInput}
                        placeholder="Text Message"
                        onChangeText={(text) => this.setState({ text })}
                        value={this.state.text}
                    />
                   <Icon
                        raised
                        name='send'
                        type='font-awesome'
                        color= {themeColor}
                        size = {20}
                        onPress={() => this.sendMessage()} 
                    />
                </View>
            </KeyboardAvoidingView>
        </React.Fragment>
        )
    }
}

const styles = {
    header:{
        backgroundColor: '#2c2638',
        height: height*0.1
      },
      headerTitle:{
        textTransform: 'capitalize',
        paddingTop:height*0.03,
        paddingBottom:50,
        fontFamily: 'BebasNeue',
        fontSize : 25,
        color: '#ffffff'
    },
  textInput: {
    height: 40,
    borderColor: 'lightgray',
    borderTopWidth: StyleSheet.hairlineWidth,
    flex: 1,
    paddingHorizontal: 8,
    marginTop: 10
  },
  sendButton: {
    width: 80,
    backgroundColor: themeColor,
    height: 40,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderColor: 'lightgray',
    marginTop: 10
  },
  sendButtonText: {
    fontSize: 16,
    color: 'white',
    lineHeight: 30,
    textAlign: 'center',
    fontWeight: 'bold'
  },
  messageContainer: {
    flexDirection: 'row',
    padding: 10
  },
  messageBody: {
    flex: 1,
    padding: 8,
    borderRadius: 5
  },
  profile: {
    width: 40,
    height: 40,
    borderRadius: 20
  },
  createdAt: {
    color: 'gray',
    fontSize: 12,
    marginHorizontal: 5
  }
}