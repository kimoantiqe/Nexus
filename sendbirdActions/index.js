export * from './user';
export * from './groupChannel'