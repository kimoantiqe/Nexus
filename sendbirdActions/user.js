import SendBird from 'sendbird';

const APP_ID = '5684560F-4764-4763-A8EB-3AD372D8B9EB';

//Connect Function.
export const sbConnect = (userId, nickname) => {
    return new Promise((resolve, reject) => {
        //Create new connection.
        const sb = new SendBird({ 'appId': APP_ID });
        //Initiate connection to our sendbird backend
        sb.connect(userId, (user, error) => {
            //Handles the error
            if (error) {
                reject('SendBird Login Failed.');
            } else {
                //Update the user profile to view his name, 
                //Second option is set to null until we implement 
                //profile pics.
                sb.updateCurrentUserInfo(nickname, null, (user, error) => {
                    if (error) {
                        reject('Update User Failed.');
                    } else {
                        resolve(user);
                    }
                })
            }
        })
    })
};

//Disconnect Function.
export const sbDisconnect = () => {
    return new Promise((resolve, reject) => {
        const sb = SendBird.getInstance();
        if (sb) {
            sb.disconnect(() => {
                resolve(null);
            });
        } else {
            resolve(null);
        }
    })
}