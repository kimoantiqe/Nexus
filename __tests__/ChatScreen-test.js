import React from 'react';
import ChatScreen from '../screens/ChatScreen'

import {sbConnect, sbDisconnect} from '../sendbirdActions'

import { shallow } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Icon } from 'react-native-elements'

import {
    Header,
    Left,
    Right,
    Body,
    Title
  } from "native-base";

import toJson from 'enzyme-to-json';

import {
    AppRegistry,
    StyleSheet,
    Text,
    View,
    Image,
    Button,
    TextInput,
    TouchableHighlight,
    FlatList,
    Dimensions,
    KeyboardAvoidingView
  } from 'react-native'
 
Enzyme.configure({ adapter: new Adapter() });

test('renders correctly', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation} />);
    expect(toJson(wrapper)).toMatchSnapshot();

    sbDisconnect();
});

test('Check if TextInput is present', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation} />);

    expect(wrapper.find(TextInput)).toHaveLength(1);

    sbDisconnect();
});

test('Check Icon onPress', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    wrapper.instance().sendMessage = sendMessageMock;
    wrapper.update();

    wrapper.find(Icon).first().props().onPress();

    expect(sendMessageMock).toHaveBeenCalled();

    sbDisconnect();

});

test('Check Text Input', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    wrapper.find(TextInput).first().props().onChangeText("Hello");

    expect(wrapper.find(TextInput).first().props().value).toEqual("Hello");

    sbDisconnect();

});

test('Check message array not updated with empty input', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    const messagesArray = wrapper.instance().state.messages;

    wrapper.find(Icon).first().props().onPress();

    expect(wrapper.instance().state.messages).toEqual(messagesArray);

    sbDisconnect();

});

test('Check send message box', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    const sender = {userId: userid};

    const msg = {_sender: sender};

    const out = wrapper.instance().renderMessage(msg);
    
    expect(out).toBeDefined();

    sbDisconnect();

});

test('Check received message box', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    const sender = {userId: "5bcf97fd4a5aa600150cc399"};

    const msg = {_sender: sender};

    const out = wrapper.instance().renderMessage(msg);

    expect(out).toBeDefined();

    sbDisconnect();

});

test('Check List containes messages', () => {

    let userid = "5bcf97fd4a5aa600150cc338";
    let nickname = "Ahmed";

    sbConnect(userid, nickname);

    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    
    //mock press event
    const sendMessageMock = jest.fn();
    sendMessageMock.mockReturnValue('Send Button Pressed');

    navigation.getParam
    .mockReturnValueOnce("sendbird_group_channel_86468035_24ca64b3b280f3552d5c287d8ae2fcf40d238388")
    .mockReturnValueOnce(userid);
    
    const wrapper = shallow(<ChatScreen navigation={navigation}/>);

    const msg = ["Hello", "One"];

    wrapper.instance().state.messages = msg;
    wrapper.update();

    expect(wrapper.instance().state.messages).toEqual(msg);

    sbDisconnect();

});
