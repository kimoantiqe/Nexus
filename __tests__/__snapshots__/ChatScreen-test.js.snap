// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Styled(Header)
    androidStatusBarColor="#ffffff"
    iosBarStyle="light-content"
    style={
      Object {
        "backgroundColor": "#2c2638",
        "height": 133.4,
      }
    }
  >
    <Styled(Left) />
    <Styled(Body)>
      <Styled(Title)
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "BebasNeue",
            "fontSize": 25,
            "paddingBottom": 50,
            "paddingTop": 40.019999999999996,
            "textTransform": "capitalize",
          }
        }
      />
    </Styled(Body)>
    <Styled(Right) />
  </Styled(Header)>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    inverted={true}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "flex": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={0}
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Text Message"
        style={
          Object {
            "borderColor": "lightgray",
            "borderTopWidth": 0.5,
            "flex": 1,
            "height": 40,
            "marginTop": 10,
            "paddingHorizontal": 8,
          }
        }
        value=""
      />
      <Icon
        color="#2f1959"
        name="send"
        onPress={[Function]}
        raised={true}
        reverse={false}
        reverseColor="white"
        size={20}
        type="font-awesome"
        underlayColor="white"
      />
    </Component>
  </KeyboardAvoidingView>
</Fragment>
`;
